include/openmrn_features.h

```diff
@@ -101,6 +103,39 @@
 #endif
 #endif
 
+#if defined(__linux__) || defined(__MACH__) || defined(__FreeRTOS__) ||        \
+    defined(ESP32)
+/// Compiles support for BSD sockets API.
+#define OPENMRN_FEATURE_BSD_SOCKETS 1
+
+#ifndef __FreeRTOS__
+/// Compiles support for setting the SO_RCVTIMEO value when creating a new
+/// socket.
+#define OPENMRN_HAVE_BSD_SOCKETS_RX_TIMEOUT 1
+#endif
+
+#if !defined(__FreeRTOS__) && !defined(ESP32)
+/// Compiles support for calling getsockname when binding a socket to a port
+/// when listening for incoming connections.
+#define OPENMRN_HAVE_BSD_SOCKETS_GETSOCKNAME 1
+#endif
+
+#if defined(__linux__)
+/// Compiles suport for IPv6 address types
+#define OPENMRN_HAVE_BSD_SOCKETS_IPV6 1
+#endif
+
+#if defined(__linux__)
+/// Compiles suport for ignoring SIGPIPE write failures on sockets.
+#define OPENMRN_BSD_SOCKETS_IGNORE_SIGPIPE 1
+#endif
+
+#if defined(__linux__) || defined(ESP32)
```

Comment:
`__MACH__` ?

Reply:
I think it makes sense but I don't have a MacOS available to test with. I'll add it here and we can remove it if we find it causes issues.